src/Jellyfin.LiveTv/Guide/GuideManager.cs

```diff
@@ -0,0 +1,709 @@
+    {
+        ArgumentNullException.ThrowIfNull(progress);
+
+        await EmbyTV.EmbyTV.Current.CreateRecordingFolders().ConfigureAwait(false);
+
+        await _tunerHostManager.ScanForTunerDeviceChanges(cancellationToken).ConfigureAwait(false);
+
+        var numComplete = 0;
+        double progressPerService = _liveTvManager.Services.Count == 0
+            ? 0
+            : 1.0 / _liveTvManager.Services.Count;
+
+        var newChannelIdList = new List<Guid>();
+        var newProgramIdList = new List<Guid>();
+
+        var cleanDatabase = true;
+
+        foreach (var service in _liveTvManager.Services)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            _logger.LogDebug("Refreshing guide from {Name}", service.Name);
+
+            try
+            {
+                var innerProgress = new ActionableProgress<double>();
+                innerProgress.RegisterAction(p => progress.Report(p * progressPerService));
+
+                var idList = await RefreshChannelsInternal(service, innerProgress, cancellationToken).ConfigureAwait(false);
+
+                newChannelIdList.AddRange(idList.Item1);
+                newProgramIdList.AddRange(idList.Item2);
+            }
+            catch (OperationCanceledException)
+            {
+                throw;
+            }
+            catch (Exception ex)
+            {
+                cleanDatabase = false;
+                _logger.LogError(ex, "Error refreshing channels for service");
+            }
+
+            numComplete++;
+            double percent = numComplete;
+            percent /= _liveTvManager.Services.Count;
+
+            progress.Report(100 * percent);
+        }
+
+        if (cleanDatabase)
+        {
+            CleanDatabase(newChannelIdList.ToArray(), [BaseItemKind.LiveTvChannel], progress, cancellationToken);
+            CleanDatabase(newProgramIdList.ToArray(), [BaseItemKind.LiveTvProgram], progress, cancellationToken);
+        }
+
+        var coreService = _liveTvManager.Services.OfType<EmbyTV.EmbyTV>().FirstOrDefault();
+        if (coreService is not null)
+        {
+            await coreService.RefreshSeriesTimers(cancellationToken).ConfigureAwait(false);
+            await coreService.RefreshTimers(cancellationToken).ConfigureAwait(false);
+        }
+
+        progress.Report(100);
+    }
+
+    private double GetGuideDays()
+    {
+        var config = _config.GetLiveTvConfiguration();
+
+        return config.GuideDays.HasValue
+            ? Math.Clamp(config.GuideDays.Value, 1, MaxGuideDays)
+            : 7;
+    }
+
+    private async Task<Tuple<List<Guid>, List<Guid>>> RefreshChannelsInternal(ILiveTvService service, ActionableProgress<double> progress, CancellationToken cancellationToken)
+    {
+        progress.Report(10);
+
+        var allChannelsList = (await service.GetChannelsAsync(cancellationToken).ConfigureAwait(false))
+            .Select(i => new Tuple<string, ChannelInfo>(service.Name, i))
```

Comment:
This seems wasteful, no need to create a Tuple here, right? We can just use `service.Name` directly in the following foreach loop